Assets/_BForBoss/_Weapons/Scripts/Weapons/Weapon.cs

```diff
@@ -51,12 +51,18 @@ public Weapon(IWeaponProperties weaponProperties, IRandomUtility randomUtility =
              _ammunitionAmount = weaponProperties.AmmunitionAmount;
          }
 
-         public void DecrementElapsedTimeRateOfFire(float deltaTime)
+         public void DecrementElapsedTimeRateOfFire(float deltaTime, float timeScale)
          {
-             _elapsedRateOfFire = Mathf.Clamp(_elapsedRateOfFire - deltaTime, 0, float.PositiveInfinity);
+             var scaledDeltaTime = ScaledDeltaTime(deltaTime, timeScale);
+             _elapsedRateOfFire = Mathf.Clamp(_elapsedRateOfFire - scaledDeltaTime, 0, float.PositiveInfinity);
          }
 
-         public void ReloadWeaponCountDownIfNeeded(float deltaTime)
+         public float ScaledDeltaTime(float deltaTime, float timeScale)
+         {
+             return 1 / timeScale * deltaTime;
```

Comment:
Just for confirmation,  both `deltaTime` and `timeScale` can never be 0 right? would it be better to have a check before the division? 

I'm down either way

Reply:
Good point. I'll clamp it.